### ISSUE TITLE
UnitOfWork <-> DefaultDbCommand implementation coupling

### ISSUE THREAD

**Issue by pchemguy** [pchemguy/SecureADODB-Fork#5]
_UnitOfWork_ class is coupled to a particular implementation of _DefaultDbCommand_ class, a particular implementation of the _IDbCommand_ interface:
```
   UnitOfWork.FromConnectionString ->
-> DefaultDbCommandFactory.Create ->
-> DefaultDbCommandFactory.IDbCommandFactory_Create ->
-> DefaultDbCommand.Create
```
Ideally, _UnitOfWork.FromConnectionString_ should, possibly, take an additional parameter, specifying the desirable implementation of the _IDbCommand_ interface. Additionally, it would also require a decoupled implementation of the _IDbCommandFactory_ interface.

Depends on issue IDbCommandFactory https://github.com/pchemguy/RDVBA-examples/issues/6
[Branch Ref](https://github.com/pchemguy/RDVBA-examples/tree/SecureADODB/DbCommandFactory) - merged into [Branch Ref](https://github.com/pchemguy/RDVBA-examples/tree/SecureADODB/codebase-refactoring)

Alternative implementation - DbManager: commits https://github.com/pchemguy/RDVBA-examples/commit/8749649e8a316c00457f8e7c06ab378c56905ce6 and https://github.com/pchemguy/RDVBA-examples/commit/a3fc89690274da4942d74d4ccdf04721893d3325
